src/main/java/za/ac/sun/cs/search/singleagent/IDAStarAgent.java

```diff
@@ -35,7 +33,7 @@ public IDAStarAgent(short[] configuration) {
             bound = i;
         }               
 
-        return path.toArray(new Direction[path.size()]);
+        return path.toArray(new Direction[0]);
```

Comment:
`toArray` [apparently](https://docs.oracle.com/javase/7/docs/api/java/util/Vector.html#toArray) figures out size for us.

Reply:
Cool, did not know that.